glib-macros/tests/dynamic_enums.rs

```diff
@@ -0,0 +1,494 @@
+            MyModuleEnumLazy::Foo
+        );
+        assert_eq!(
+            unsafe { MyModuleEnumLazy::from_glib(1) },
+            MyModuleEnumLazy::Bar
+        );
+
+        let t = MyModuleEnumLazy::static_type();
+        assert!(t.is_a(glib::Type::ENUM));
+        assert_eq!(t.name(), "MyModuleEnumLazy");
+
+        let e = glib::EnumClass::with_type(t).expect("EnumClass::new failed");
+        let v = e.value(0).expect("EnumClass::get_value(0) failed");
+        assert_eq!(v.name(), "Foo");
+        assert_eq!(v.nick(), "foo");
+        let v = e.value(1).expect("EnumClass::get_value(1) failed");
+        assert_eq!(v.name(), "Bar");
+        assert_eq!(v.nick(), "bar");
+        assert_eq!(e.value(2), None);
+
+        // within enums registered as dynamic types, values are usables only if
+        // at least one type class ref exists (see `glib::EnumClass`).
+        assert_eq!(
+            MyModuleEnumLazy::Foo.to_value().get::<MyModuleEnumLazy>(),
+            Ok(MyModuleEnumLazy::Foo)
+        );
+        assert_eq!(
+            MyModuleEnumLazy::Bar.to_value().get::<MyModuleEnumLazy>(),
+            Ok(MyModuleEnumLazy::Bar)
+        );
+
+        // simulates the glib type system to unload the module.
+        TypeModuleExt::unuse(module);
+    }
+}
+
+pub mod plugin {
+    use super::*;
+
+    pub mod imp {
+        use glib::EnumClass;
+
+        use super::*;
+        use std::cell::Cell;
+
+        // impl for a type plugin (must implement `glib::TypePlugin`).
+        #[derive(Default)]
+        pub struct MyPlugin {
+            my_enum_type_values: Cell<Option<&'static [glib::EnumValue]>>,
+            my_enum_lazy_type_values: Cell<Option<&'static [glib::EnumValue]>>,
+        }
+
+        #[glib::object_subclass]
+        impl ObjectSubclass for MyPlugin {
+            const NAME: &'static str = "MyPlugin";
+            type Type = super::MyPlugin;
+            type Interfaces = (glib::TypePlugin,);
+        }
+
+        impl ObjectImpl for MyPlugin {}
+
+        impl TypePluginImpl for MyPlugin {
+            fn use_plugin(&self) {
+                // register enums as dynamic types.
+                let my_plugin = self.obj();
+                super::MyPluginEnum::on_implementation_load(my_plugin.as_ref());
+                super::MyPluginEnumLazy::on_implementation_load(my_plugin.as_ref());
+            }
+
+            fn unuse_plugin(&self) {
+                // marks enums as unregistered.
+                let my_plugin = self.obj();
+                super::MyPluginEnumLazy::on_implementation_unload(my_plugin.as_ref());
+                super::MyPluginEnum::on_implementation_unload(my_plugin.as_ref());
+            }
+
+            fn complete_type_info(
+                &self,
+                type_: glib::Type,
+            ) -> (glib::TypeInfo, glib::TypeValueTable) {
+                let enum_type_values = match type_ {
+                    type_ if type_ == super::MyPluginEnum::static_type() => {
```

Comment:
Same question here. What happens if the user forgets this and then the enum is tried to be used?